core/ledger/kvledger/txmgmt/pvtstatepurgemgmt/snapshot_purge_mgr_builder_test.go

```diff
@@ -0,0 +1,182 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package pvtstatepurgemgmt
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/hyperledger/fabric/core/ledger/internal/version"
+	"github.com/hyperledger/fabric/core/ledger/kvledger/bookkeeping"
+	"github.com/hyperledger/fabric/core/ledger/pvtdatapolicy"
+	btltestutil "github.com/hyperledger/fabric/core/ledger/pvtdatapolicy/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+func TestPurgeMgrBuilder(t *testing.T) {
+	bookkeepingEnv := bookkeeping.NewTestEnv(t)
+	defer bookkeepingEnv.Cleanup()
+
+	ledgerID := "test-ledger"
+	bookkeepingProvider := bookkeepingEnv.TestProvider
+	btlPolicy := btltestutil.SampleBTLPolicy(
+		map[[2]string]uint64{
+			{"ns1", "coll1"}: 1,
+			{"ns2", "coll2"}: 2,
+		},
+	)
+
+	purgeMgrBuilder := NewPurgeMgrBuilder(ledgerID, btlPolicy, bookkeepingProvider)
+	// add data that expires at block-7
+	require.NoError(t,
+		purgeMgrBuilder.ConsumeSnapshotData("ns1", "coll1", []byte("key-hash-1"), version.NewHeight(5, 1).ToBytes()),
+	)
+	require.NoError(t,
+		purgeMgrBuilder.ConsumeSnapshotData("ns1", "coll1", []byte("key-hash-2"), version.NewHeight(5, 1).ToBytes()),
+	)
+	// add data that expires at block-8
+	require.NoError(t,
+		purgeMgrBuilder.ConsumeSnapshotData("ns2", "coll2", []byte("key-hash-2"), version.NewHeight(5, 1).ToBytes()),
+	)
+	// add data that never expires
+	require.NoError(t,
+		purgeMgrBuilder.ConsumeSnapshotData("ns1", "no-btl-collection", []byte("key-hash-3"), version.NewHeight(5, 1).ToBytes()),
```

Comment:
If `GetExpiringBlock()` could not find the collection, wouldn't it return `NoSuchCollectionError`? Why isn't it returning error here?